lib/cylc/task_proxy.py

```diff
@@ -332,23 +329,41 @@ def __init__(
 
     def _add_prerequisites(self, point):
         """Add task prerequisites."""
-        # NOTE: Task objects hold all triggers defined for the task
-        # in all cycling graph sections in this data structure:
-        #     self.triggers[sequence] = [list of triggers for this
-        #     sequence]
-        # The list of triggers associated with sequenceX will only be
-        # used by a particular task if the task's cycle point is a
-        # valid member of sequenceX's sequence of cycle points.
-
-        # 1) non-conditional triggers
-        ppre = plain_prerequisites(self.identity, self.tdef.start_point)
-        spre = plain_prerequisites(self.identity, self.tdef.start_point)
+        # self.triggers[sequence] = [triggers for sequence]
+        # Triggers for sequence_i only used if my cycle point is a
+        # valid member of sequence_i's sequence of cycle points.
+
+        for sequence in self.tdef.triggers.keys():
+            for ctrig, exp in self.tdef.triggers[sequence]:
```

Comment:
Is it possible to do this to avoid a search for each key:

```python
for sequence, exps in self.tdef.triggers.items():
    for ctrig, exp in exps:
        # ...
```

Otherwise, I believe you can just do:

```python
for sequence in self.tdef.triggers:
    for ctrig, exp in self.tdef.triggers[sequence]:
        # ...
```